test: allow disabling crypto tests

codebytere opened the pull request:
This PR adds a new environment variable to the test suit: `NODE_SKIP_CRYPTO`. Right now whether or not crypto tests can be run is solely determined by whether or not `openssl` is defined on the `process` object, which is true for Electron.js since many consumers use it to determine the presence of crypto even though we use BoringSSL and thus experience incompatibilities.

This new option would allow us to more effectively run our own smoke tests of Node.js' test suite without needing to manually disable every crypto test individually.

cc @tniessen 

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

<!--
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

Trott commented:
With this change, using `--no-crypto` doesn't cause any tests to be skipped as far as I can tell. Am I doing something wrong? I'm trying things like `tools/test.py --no-crypto -p tap async-hooks` and seeing if the crypto tests in the `async-hooks` directory are skipped.

codebytere commented:
@Trott It turns out that individual tests are validated in js with the same mechanism, so i've moved this logic into common and added documentation for a new simple env var, `NODE_SKIP_CRYPTO`. I'm happy to name it something else if you all would prefer :)

cc @addaleax since i've moved this mechanism, this probably needs a re-review

BridgeAR commented:
Marking as author ready without CI, since it should not have any effect.

codebytere commented:
@addaleax it turned out that it was used but that it's purpose wasn't clear at first glance; it's used [here](https://github.com/nodejs/node/blob/2ebd1a0d3f22a3743f11f0d6bbc8b1177194c9b3/test/testpy/__init__.py#L76-L80)

richardlau commented:
> 
> 
> Marking as author ready without CI, since it should not have any effect.

I'd still suggest running CI to make sure, for example, ubuntu1804_sharedlibs_withoutssl_x64 still works.

Trott commented:
Landed in be055d103c85
